pkg/network/render.go

```diff
@@ -886,6 +908,110 @@ func renderIPTablesAlerter(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.
 	return manifests, nil
 }
 
+// renderNetworkingConsolePlugin renders the common objects related to the networking console plugin resources
+func renderNetworkingConsolePlugin(manifestDir string, cl cnoclient.Client) ([]*uns.Unstructured, error) {
+	if !isConsolePluginCRDExist(cl) {
+		log.Printf("consoleplugins.console.openshift.io CRD does not exist yet")
+		return nil, nil
+	}
+
+	tmpFallbackConsoleImage := "quay.io/orenc/networking-console-plugin:4.16"
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+
+	val, ok := os.LookupEnv("NETWORKING_CONSOLE_PLUGIN_IMAGE")
+	if ok {
+		data.Data["NetworkingConsolePluginImage"] = val
+	} else {
+		data.Data["NetworkingConsolePluginImage"] = tmpFallbackConsoleImage
+	}
+
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "networking-console-plugin"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render networking-console-plugin manifests")
+	}
+	return manifests, nil
+}
+
+type Patch struct {
+	Type types.PatchType
+	Data []byte
+}
+
+// registerNetworkingConsolePlugin enables console plugin for networking-console if not already enabled
+func registerNetworkingConsolePlugin(cl cnoclient.Client) error {
+	if !isConsolePluginCRDExist(cl) {
+		return nil
+	}
+	pluginName := "networking-console-plugin"
+	clusterConsole := "cluster"
+	consoleKey := client.ObjectKey{Name: "cluster"}
+	consoleObj := &operv1.Console{}
+
+	crc := cl.Default().CRClient()
+	err := crc.Get(context.TODO(), consoleKey, consoleObj)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("NOTICE: consoles.operator.openshift.io %s is not available yet.", clusterConsole)
+			return nil
+		}
+		return errors.Wrapf(err, "Could not get consoles.operator.openshift.io resource")
+	}
+
+	if stringInSlice(consoleObj.Spec.Plugins, pluginName) {
+		return nil
+	}
+
+	var patches []cnoclient.JsonPatch
+
+	if consoleObj.Spec.Plugins == nil {
```

Comment:
Please use applyconfigurations instead of a raw JsonPath. You might even skip the `Get` to only do one API call.